exercises/07-normalizr/src/modules/users/users-saga.ts

```diff
@@ -0,0 +1,59 @@
+import { all, call, fork, put, takeEvery } from 'redux-saga/effects'
+import { AxiosResponse } from 'axios'
+
+import * as Schema from 'modules/entities/entities-schema'
+import * as EntitiesTypes from 'modules/entities/entities-types'
+
+import { usersActions } from 'modules/users/users-slice'
+import * as UsersEffects from 'modules/users/users-effects'
+import { normalizeAndStore } from 'modules/entities/entities-saga'
+import { User } from './user-types'
+
+const logError = (error: unknown) => {
+  if (error instanceof Error) {
+    console.log(error.message)
+  } else {
+    console.log(error)
+  }
+}
+
+function* getUsers() {
+  let normalizedUsers: EntitiesTypes.UserIds = []
+
+  try {
+    const { data: users }: AxiosResponse<User[]> = yield call(
+      UsersEffects.getUsers
+    )
+
+    normalizedUsers = yield normalizeAndStore<
+      EntitiesTypes.User[],
+      EntitiesTypes.UserEntities,
+      EntitiesTypes.UserIds
+    >(users, Schema.users)
```

Comment:
Why can't we use `yield call`? Due to the generic arguments?

Reply:
Yes. Is it somehow possible to use `call` with these generic arguments? I'm not at all sure with normalizr typings here. If I remove typings completely everything still works without errors and we could use `call`. Do we actually need those typings here?

Reply:
> Is it somehow possible to use call with these generic arguments?

I'm not aware of any way with the current typing of `call`.

> Do we actually need those typings here?

I don't think it brings any value here. There are 3 arguments in the `normalizr` typing. `E` and `R` are used only for the output which means they are totally useless in type-safety. `T` is for the input but I guess it can be anything and it's not used in the output anyway, so I can't see anything useful here.

Reply:
Removed generic arguments 8f435ac

Reply:
@novellizator as agreed, removed additionally schema typings 62750fbf5651b1cceb731432d5fc437028bfccdf